### PR TITLE
Add the ability to specify an S3 bucket policy.

### PR DESCRIPTION
**Pull request by rererecursive**

### CODE REVIEW

File: s3-cloudfront.config.yaml [hunk 1]
```diff
@@ -1 +1,5 @@
-bucket_name: frontend.${EnvironmentName}.${DnsDomain}
\ No newline at end of file
+bucket_name: frontend.${EnvironmentName}.${DnsDomain}
+bucket_policy:
```
Comment (aaronwalker) [inline s3-cloudfront.config.yaml]: I think the bucket_policy shouldn't be enabled by default. Just put this block in but comment it out as an example and implement a test for this config. See the https://github.com/theonestack/hl-component-cloudfront/tree/master/tests as an example for creating a test

**Comment by Samseppiol** [comment]
@rererecursive @aaronwalker @Guslington Would it be worth using the s3 component instead which already supports this?

**Comment by Guslington** [comment]
@Samseppiol yes probably should be referencing the s3 component if more of the features from that component are required by this component.

**Comment by rererecursive** [comment]
@Samseppiol Yes, however trying to integrate the existing S3 component is turning out to be a fairly hacky job, as the former expects a config containing a list of buckets with their own configs, whereas this component expects just a single bucket name. 
We'd also have to apply a similar fix to the nested component name as we did with the CloudFront component, e.g.:
`Component name: s3_component_name, template: 's3', render: Inline` 
Working around these without causing breaking changes seems either messy or difficult..

**Pull request merged**